### DISCUSSION TITLE
Enhancing metadata embedding in records

### DISCUSSION THREAD

**Discussion by Ndpnt** [OpenTermsArchive/engine#1060] [Use case research]
Hi everyone,

As Open Terms Archive continues to evolve, we're currently confronted to the question of which metadata should be attached to records and in which way. So we're opening a discussion on this subject to best adapt the record metadata handling to serve the diverse needs of the project's various environments.

We're curious to hear about any challenges you've encountered or any specific needs you've identified regarding metadata. Which information you've found lacking and in which contexts, for which usages?

In our side, for instance, during our research on how to recover from errors in the version history, we realized the value of embedding engine version as technical metadata for better traceability. This metadata for example could allow to fix errors in history that was only present on some engine versions.

Additionally, it seems that some of you have already have encounter the need to add the terms location at the beginning of versions, as seen in [this example](https://code.peren.fr/hackathon-2024/retrieval-modules/ota-engine/-/merge_requests/1/diffs.).
It could be interesting to know what was the need here?

From these two examples, it appears that there are two primary types of metadata: **technical metadata**, which describe the technical aspects of a record (for example: engine version, browser used to fetch the record, geolocation of the server, …), and **content metadata**, which offer context and additional information about the content itself (for example: URL where the content was fetched, selectors used to get the content, …). 
We believe that these two kinds of metadata could be very useful and they probably be handle differently to optimize uses.

So, we invite you to share your needs and your thoughts about records metadata!

**Comment by LVerneyEC** [comment]
Hi,

I have a use case for this and referencing #1183 for initial discussion. This is especially useful when reviewing the versions and assessing whether this is a substantial change or not, to be able to quickly go from the recorded version to the original URL. Also important sometimes to assess whether the change is correctly reflected and captured on the upstream site or whether this might be a data collection artifact (geolocation, issue with selectors, etc.). 

I'd also say this is even more important for multi-page documents, for which it might come tricky to find the relevant URL for a given section of version.
